css/amp.css

```diff
@@ -153,6 +153,14 @@ html.i-amphtml-ios-embed.i-amphtml-ios-overscroll > #i-amphtml-wrapper {
   position: relative;
 }
 
+/**
+ * Children of responsive elements should not be able to influence the height
+ * which they do when they are statically positioned.
+ */
+.i-amphtml-layout-responsive > :not(i-amphtml-sizer) {
```

Comment:
The children of not-built elements here should be `display:none` - so they can never overflow. This should be true for all size-defining elements. The CSS that Ali is referencing should work. There's no phase that'd set `i-amphtml-layout-responsive` but not `i-amphtml-notbuilt`. As a matter of fact, `i-amphtml-notbuilt` is the first class that's set.